src/ExpressionEvaluator/Core/Source/ExpressionCompiler/InstructionDecoder.cs

```diff
@@ -22,6 +20,18 @@ internal abstract class InstructionDecoder<TCompilation, TMethodSymbol, TModuleS
             memberOptions: SymbolDisplayMemberOptions.IncludeContainingType | SymbolDisplayMemberOptions.IncludeExplicitInterface,
             miscellaneousOptions: SymbolDisplayMiscellaneousOptions.UseSpecialTypes);
 
+        private readonly bool _useReferencedAssembliesOnly;
+
+        internal InstructionDecoder()
+        {
+            _useReferencedAssembliesOnly = ExpressionCompiler.GetUseReferencedAssembliesOnlySetting();
```

Comment:
>ExpressionCompiler.GetUseReferencedAssembliesOnlySetting(); [](start = 43, length = 59)

Shouldn't this rather flow from the ExpressionCompiler instance?

Reply:
Yes, ideally. Added a comment for now.

---
In reply to: [181508668](https://github.com/dotnet/roslyn/pull/25955#discussion_r181508668) [](ancestors = 181508668)